### DISCUSSION TITLE
Unique decision variables

### DISCUSSION THREAD

**Discussion by KadircanKara** [anyoptimization/pymoo#306] [Q&A]
Hello all. I am looking for a way to enforce unique decision variables in Pymoo. Any help would be very appreciated. Thanks :)

**Comment by ubeydtalha** [comment]
I do not understand exactly you want . You can ask me Turkish ,maybe I can help.

**Reply by KadircanKara** [reply]
Selamlar. Değişkenlerim tam sayı ve her değişkenin farklı değerde olması lazım. Hedef fonksiyonlarımın birinde mTSP'yi çözmeye çalışıyorum ve değişkenlerim lokasyon (şehir) numaralarını oluşturacak ve her lokasyonun sadece bir kere ziyaret edilmesi gerektiği için böyle bir zorunluluk var.

Umarım anlatabilmişimdir. Teşekkürler.

**Reply by ubeydtalha** [reply]
Kütüphanenin resmi sitesinde TSP problemi için bir örnek var.
https://pymoo.org/customization/permutation.html

Ayrıca yeni versiyonda tam sayı değerleri için şöyle bir yol var.
https://pymoo.org/customization/mixed.html
